### PR TITLE
Documented a few more Pallas APIs and added them to the API docs

### PR DESCRIPTION
**Pull request by superbobry**

### CODE REVIEW

File: jax/_src/pallas/primitives.py [hunk 1]
```diff
@@ -170,41 +171,96 @@ def _atomic_abstract_eval(*avals_flat, args_tree, atomic_type: AtomicOpType):
...
+
+def atomic_add(x_ref_or_view, idx, val, *, mask: Any | None = None):
+  """Atomically adds a given value to the value at the given index."""
+  return _atomic_rmw(
+      x_ref_or_view, idx, val, mask=mask, atomic_type=AtomicOpType.ADD
+  )
+
+
+def atomic_max(x_ref_or_view, idx, val, *, mask: Any | None = None):
+  """Atomically computes the maximum of the given value and the value at the
```
Comment (gnecula) [inline jax/_src/pallas/primitives.py:197]: nit: not just computes, but I assume it stores the maximum in the ref?

same for `atomic_min` below.

File: jax/_src/pallas/primitives.py [hunk 2]
```diff
@@ -170,41 +171,96 @@ def _atomic_abstract_eval(*avals_flat, args_tree, atomic_type: AtomicOpType):
...
 
 def atomic_cas(ref, cmp, val):
+  """Performs an atomic compare-and-swap of the value in the ref with the
+  given value.
+
+  Args:
+    ref: The ref to operate on.
+    cmp: The expected value to compare against.
+    val: The value to swap in.
+  """
```
Comment (gnecula) [inline jax/_src/pallas/primitives.py:263]: Document the return value?

File: jax/_src/pallas/primitives.py [hunk 3]
```diff
@@ -493,11 +548,24 @@ def _swap_discharge_rule(in_avals, out_avals, *args_flat, args_tree, **_):
...
+  Args:
+    x_ref_or_view: The ref to load from.
+    idx: The indexer to use.
+    mask: An optional boolean mask specifying which indices to load.
+      If mask is ``False`` and ``other`` is not given, no assumptions can
+      be made about the value in the resulting array.
+    other: An optional value to use for indices where mask is ``False``.
+    cache_modifier: TO BE DOCUMENTED.
+    eviction_policy: TO BE DOCUMENTED.
+    volatile: TO BE DOCUMENTED.
```
Comment (gnecula) [inline jax/_src/pallas/primitives.py:567]: Document the returned value?

Comment (gnecula) [reply jax/_src/pallas/primitives.py:567]: Actually, it is pretty obvious what that would be, but it is not bad habit to add a line to the docstring about it.

**Review by gnecula — APPROVED** [review]
Thanks!

File: jax/_src/pallas/primitives.py (hunk 1 above)
Comment (superbobry) [reply jax/_src/pallas/primitives.py:197]: Good point. Updated the docs. PTAL.

File: jax/_src/pallas/primitives.py (hunk 2 above)
Comment (superbobry) [reply jax/_src/pallas/primitives.py:263]: Done.

File: jax/_src/pallas/primitives.py (hunk 3 above)
Comment (superbobry) [reply jax/_src/pallas/primitives.py:567]: Rephrased the docstring to start with "Returns ...".

**Pull request merged**